src/overlays/actors/ovl_Bg_Spout_Fire/z_bg_spout_fire.c

```diff
@@ -31,39 +34,172 @@ const ActorInit Bg_Spout_Fire_InitVars = {
-    { ELEMTYPE_UNK0, { 0x20000000, 0x01, 0x04 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_ON | TOUCH_SFX_NONE, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_PLAYER,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x20000000, 0x01, 0x04 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NONE,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 30, 83, 0, { 0, 0, 0 } },
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit D_80A6118C = { 1, 80, 100, MASS_IMMOVABLE };
+static CollisionCheckInfoInit sColChkInfoInit = { 1, 80, 100, MASS_IMMOVABLE };
 
-#endif
-
-extern ColliderCylinderInit D_80A61160;
-extern CollisionCheckInfoInit D_80A6118C;
-
-extern UNK_TYPE D_06000040;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/BgSpoutFire_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/BgSpoutFire_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60C24.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60C94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60CDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60D10.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60DA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60E08.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/BgSpoutFire_Update.s")
+static TexturePtr D_80A61194[] = {
+    object_fwall_Tex_003CA0, object_fwall_Tex_0040A0, object_fwall_Tex_0044A0, object_fwall_Tex_0048A0,
+    object_fwall_Tex_004CA0, object_fwall_Tex_0050A0, object_fwall_Tex_0054A0, object_fwall_Tex_0058A0,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A61040.s")
+static s32 texturesDesegmented = 0;
+
+void BgSpoutFire_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 i;
+    BgSpoutFire* this = THIS;
+
+    this->actor.scale.z = 0.13499999f;
+    this->actor.scale.x = 0.13499999f;
+    this->actor.scale.y = 0.01f;
+    this->unk_14A = 0;
+    Collider_InitAndSetCylinder(globalCtx, &this->unk14C, &this->actor, &sCylinderInit);
+    this->unk14C.dim.pos.y = this->actor.world.pos.y;
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    if (!texturesDesegmented) {
+        for (i = 0; i < ARRAY_COUNT(D_80A61194); i++) {
+            D_80A61194[i] = Lib_SegmentedToVirtual(D_80A61194[i]);
+        }
+        texturesDesegmented = true;
+    }
+    this->actionFunc = func_80A60C94;
+}
+
+void BgSpoutFire_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpoutFire* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->unk14C);
```

Comment:
```suggestion
    Collider_DestroyCylinder(globalCtx, &this->collider);
```
Should just be able to rename unk14C to collider